src/v/kafka/server/handlers/metadata.cc

```diff
@@ -322,7 +322,7 @@ guess_peer_listener(request_context& ctx, cluster::broker_ptr broker) {
       ctx.listener());
 
     // Look up port for the listener in use for this request
-    const auto& my_listeners = config::node().advertised_kafka_api();
+    const auto my_listeners = config::node().advertised_kafka_api();
```

Comment:
wow nice find. i guess it's pointing to a temporary? i wonder if advertised_kafka_api used to return a const ref and that changed leading to this sisue.